File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Collection/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Collection
{
    public class MyStack<T> : MyIStack<T>
    {
        private Stack<T> stk;

        public MyStack(Stack<T> stk)
        {
            this.stk = stk;
        }

        public bool IsEmpty()
        {
            return this.stk.Count == 0;
        }

        public T Peek()
        {
            return this.stk.Peek();
        }

        public T Pop()
        {
            return this.stk.Pop();
        }

        public void Push(T element)
        {
            this.stk.Push(element);
        }

        public Stack<T> ToStack()
        {
            return this.stk;
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            this.stk.ToList().ForEach(e => sb.Append(e.ToString() + "\n"));
            return sb.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/Collection/Stack/MyStack.java
package Collection.Stack;

import java.util.List;
import java.util.Stack;

public class MyStack<T> implements MyIStack<T> {
    private Stack<T> stack;

    public MyStack(){
        stack = new Stack<T>();
    }

    @Override
    public T pop(){
        return stack.pop();
    }

    @Override
    public void push(T v)
    {
        stack.push(v);
    }

    @Override
    public boolean isEmpty()
    {
        return stack.empty();
    }

    @Override
    public String toString(){
        return stack.toString();
    }

    @Override
    public List<T> getAll() {
        return stack;
    }
}

